nyaa/search.py

```diff
@@ -3,15 +3,99 @@
 import math
 import json
 import shlex
+from datetime import datetime, timedelta
 
 from nyaa import app, db
-from nyaa import models
+from nyaa import models, torrents
 
 import sqlalchemy_fulltext.modes as FullTextMode
 from sqlalchemy_fulltext import FullTextSearch
 from elasticsearch import Elasticsearch
 from elasticsearch_dsl import Search, Q
 
+class ElasticTorrentStatsAccessor:    
+    def __init__(self, elasticResult):
+        self.seed_count = elasticResult.seed_count
+        self.leech_count = elasticResult.leech_count
+        self.download_count = elasticResult.download_count
+
+class ElasticTorrentAccessor:
+    knownAttributes = ['id', 'has_torrent', 'display_name', 'filesize', 'anonymous', 'trusted', 'remake', 'complete', 'hidden', 'deleted']
+
+    def __init__(self, elasticResult):
+        self.elastic = elasticResult
+
+    def __getattr__(self, name):
+        # Passthru for attributes that are the same
+        if name in self.knownAttributes:
+            return self.elastic[name]
+
+        # Safetynet: throw if an attribute is accessed which we don't implement    
+        raise Exception("Unknown attribute {}".format(name))
```

Comment:
@Taloth Is it really necessary to only allow attributes in `knownAttributes`?

For example this method would be more flexible and more pythonic written as follows:

```python
def __getattr__(self, item):
    try:
        return self.elastic[item]
    except KeyError:
        # Raise an AttributeError if the key does not exist
        raise AttributeError(
            '{class_name!r} object has no attribute {attr!r}'.format(
                class_name=self.__class__.__name__,
                attr=item,
            )
        )
```
It raises a more appropriate exception (`AttributeError`) with a message that mimics the default python error message.  It allows for any attribute of `self.elastic` to be accessed directly as an attribute of `ElasticTorrentAccessor` thus allowing it to be flexible and extensible.